src/features/mm_bone_data_feature.cpp

```diff
@@ -16,35 +22,17 @@ int64_t MMBoneDataFeature::get_dimension_count() const {
 
 PackedFloat32Array MMBoneDataFeature::bake_animation_pose(Ref<Animation> p_animation, double time) const {
     PackedFloat32Array result;
+
     for (int64_t i = 0; i < bone_names.size(); ++i) {
-        const String bone_path = String(_skeleton_path) + String(":") + bone_names[i];
-
-        const int32_t pos_track = p_animation->find_track(bone_path, Animation::TrackType::TYPE_POSITION_3D);
-        const int32_t rot_track = p_animation->find_track(bone_path, Animation::TrackType::TYPE_ROTATION_3D);
-
-        if (pos_track != -1) {
-            const Vector3 pos = p_animation->position_track_interpolate(pos_track, time);
-            result.append(pos.x);
-            result.append(pos.y);
-            result.append(pos.z);
-        } else {
-            result.append(0.0f);
-            result.append(0.0f);
-            result.append(0.0f);
-        }
-
-        if (rot_track != -1) {
-            const Quaternion rot = p_animation->rotation_track_interpolate(rot_track, time);
-            result.append(rot.x);
-            result.append(rot.y);
-            result.append(rot.z);
-            result.append(rot.w);
-        } else {
-            result.append(0.0f);
-            result.append(0.0f);
-            result.append(0.0f);
-            result.append(0.0f);
-        }
+        BoneState bone_state = _sample_bone_state(p_animation, time, bone_names[i]);
+
+        result.append(bone_state.pos.x);
+        result.append(bone_state.pos.y);
+        result.append(bone_state.pos.z);
+        result.append(bone_state.rot.x);
+        result.append(bone_state.rot.y);
+        result.append(bone_state.rot.z);
+        result.append(bone_state.rot.w);
```

Comment:
See the previous discussion about >5 dimensions. We need to identify a pose, such as Basis(), uniquely, but instead of a 3x3 matrix, remove three elements (any row).

Reply:
One thing to consider is that higher dimensions add more data to the dataset, and the larger the dataset the costlier the queries. In any case I'll need to read through the paper first

Reply:
The most obvious choices are:

1. Euler rotations three floats (6 possible numerical tuples for the same rotation)
2. quaternion rotations four floats (2 possible numerical tuples for the same rotation)
3. 3x3 rotations three rows (unique if normalized)
4. 2x3 rotations only two of the three rows (unique if normalized).

Reply:
https://www.zhengyiluo.com/assets/pdf/Rotation_DL.pdf